File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> mp=new HashMap();
        int n=nums.length;
        for(int i=0;i<n;i++){
            int x=target-nums[i];

            if(mp.containsKey(x)){
                return new int[]{mp.get(x), i};
            }
            mp.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> mp= new Dictionary<int,int>();
        int n=nums.Length;

        for(int i=0;i<n;i++){
            int m=target - nums[i];
            if(mp.ContainsKey(m)){
                return new int[] {mp[m],i};
            }
            mp[nums[i]]=i;
        }

        return new int[0];
    }
}